assists/baremetalconfig_xlnx.py

```diff
@@ -0,0 +1,292 @@
+#/*
+# * Copyright (c) 2020 Xilinx Inc. All rights reserved.
+# *
+# * Author:
+# *       Appana Durga Kedareswara rao <appana.durga.rao@xilinx.com>
+# *
+# * SPDX-License-Identifier: BSD-3-Clause
+# */
+import sys
+import types
+import os
+import getopt
+import re
+from pathlib import Path
+from pathlib import PurePath
+from io import StringIO
+import contextlib
+import importlib
+from lopper import Lopper
+from lopper import LopperFmt
+import lopper
+from lopper_tree import *
+from re import *
+import yaml
+
+
+def item_generator(json_input, lookup_key):
+    if isinstance(json_input, dict):
+        for k, v in json_input.items():
+            if k == lookup_key:
+                if isinstance(v, str):
+                    yield [v]
+                else:
+                    yield v
+            else:
+                for child_val in item_generator(v, lookup_key):
+                    yield child_val
+    elif isinstance(json_input, list):
+        for item in json_input:
+            for item_val in item_generator(item, lookup_key):
+                yield item_val
+
+# This API reads the schema and returns the compatible list
+def compat_list(schema):
+    if 'compatible' in schema['properties'].keys():
+        sch = schema['properties']['compatible']
+        compatible_list = []
+        for l in item_generator(sch, 'enum'):
+            compatible_list.extend(l)
+
+        for l in item_generator(sch, 'const'):
+            compatible_list.extend(l)
+
+        if 'contains' in sch.keys():
+            for l in item_generator(sch['contains'], 'enum'):
+                compatible_list.extend(l)
+
+            for l in item_generator(sch['contains'], 'const'):
+                compatible_list.extend(l)
+        compatible_list = list(set(compatible_list))
+        return compatible_list
+
+"""
+This API scans the device-tree node and returns the address
+and size of the reg property for the user provided index.
+
+Args:
+    fdt: flattened device tree represention of the dts
+    node: dtb node
+    value: property value
+    idx: index
+"""
+def scan_reg_size(fdt, node, value, idx):
+    off = fdt.parent_offset(node)
+    na = Lopper.property_get(fdt, off, "#address-cells")
```

Comment:
And these properties would be: node["#address-cells"].value[0], etc.

Reply:
Sure will fix in next pull request